packages/vite/src/node/server/index.ts

```diff
@@ -257,12 +256,13 @@ export interface ViteDevServer {
   /**
    * @internal
    */
-  _optimizeDepsMetadata: DepOptimizationMetadata | null
+  _optimizedDeps: OptimizedDeps | null
   /**
    * Deps that are externalized
    * @internal
    */
   _ssrExternals: string[] | null
+  _ssrExternalsOutdated?: boolean
```

Comment:
I can't seem to find a place where `_ssrExternalsOutdated` is set to true. Do you have plans in the future to use this variable elsewhere?

Reply:
It should be set to true here https://github.com/vitejs/vite/pull/7379/files#diff-e59380eeaeef846e1d4641a857bcdd350407f8fe96e63f54f5b218515fc9c9e4L123, but I removed it because I'm going back and forth with it. 

We started to pass `ssr` to pre-bundling as part of https://github.com/vitejs/vite/pull/3933. This was a mistake. First, the `ssr` flag will only be passed on rerun, and not on the initial scanning/optimization. And internally, the `ssr` flag isn't used. As you said before @bluwy, in SSR, we aren't using the pre-bundled dependencies. What is more, `optimizedDeps.registerMissingImport` isn't called during `ssr`, so it was never passed.

We added the code to re-calculate ssrExternals when it was added https://github.com/vitejs/vite/commit/92934d4c18329cc51879aca5016a408b467c1fa0, but as explained above this isn't really needed, as this code shouldn't be hit during ssr. Later we merged a fix because it was generating a warning during non-ssr https://github.com/vitejs/vite/pull/6492, using the flag.

So I added the `_ssrExternalsOutdated` as a way to cover both the initial commit and the warning fix... but actually, I think that we should remove this branch, and remove the new variable. If we get an issue we can add it back.

Reply:
Makes sense. Thanks for the links! I always wondered about that `ssr` thing but never looked deep into its history. It's is safe to remove it then we should do so 👍 

+1 for removing unused code if you think it won't be needed in the future. Perhaps we can do the same for `esbuildDepPlugin` and its `ssr` param too.

Reply:
Done 👍🏼